### ISSUE TITLE
Logger discards additonal context information

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/server#29358]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Use `\OCP\ILogger` or `\Psr\Log\LoggerInterface` and pass additional fields through the `$context` argument

```php
$this->logger->debug('test with data', [
    'something' => 3,
]);
```

### Expected behavior

This data ends up in the logs.

To quote https://www.php-fig.org/psr/psr-3/

> Every method accepts an array as context data. This is meant to hold any extraneous information that does not fit well in a string. The array can contain anything. Implementors MUST ensure they treat context data with as much lenience as possible. A given value in the context MUST NOT throw an exception nor raise any php error, warning or notice.

### Actual behavior

The data is silently discarded. Only `app` and `level` (and `exception` for the PSR logger) have an influence on the logged data.

As a workaround one can put as much info into the message as possibly but that has downsides
* It breaks grouping if identical errors (especially with Sentry)
* Complex structured data has to be serialized and then deserialized by hand. If it was part of the logged json a formatted log line would show the data readable out of the box

**Comment by joshtrichards** [comment]
Is this what was addressed in #32242 or is this Issue still relevant?

**Comment by juliushaertl** [comment]
Yes, let's close this.
